mitodl/ocw-studio
# Umar/2168 separately track publish date and build date

### What are the relevant tickets?
Fixes https://github.com/mitodl/ocw-studio/issues/2168

### Description (What does it do?)
This change separately tracks the build and publish dates for live/draft versions. The publish dates are updated on `mass_build` which indicates no content change. Build dates and publish dates are both updated on `publish from studio`.

### How can this be tested?
1. Switch to `umar/2168-separately-track-publish_date-and-build_date` branch `ocw-studio`
2. Check migrations applied. `docker-compose exec web ./manage.py showmigrations websites` 
    - Last applied migration should be `0057_website_draft_build_date_website_live_build_date`
    - If migration is not applied, apply migrations: `docker-compose exec web ./manage.py migrate websites`
3. Create a site in `ocw-studio` locally at `localhost:8043/sites` or use an existing one.
4. Make any changes and publish site in `ocw-studio` and Wait for the publish operation to complete. 
5. Go to Django admin at `localhost:8043/admin` and search for the site in `Websites`.
6. In Django admin verify that only `publish date` will be updated for the site.
7. Run this command to trigger a mass build locally. Replace the variables with appropriate values: 
    `docker-compose exec web ./manage.py mass_publish <live or draft> --filter <site-name>`
8. In Django admin verify that `Build date` and `Publish date` have been updated.
